client/src/components/admin_dashboard/ProfileList.js

```diff
@@ -0,0 +1,334 @@
+        } else if (props.viewing === 'STUDENT') {
+            makeCall({}, '/admin/allStudents/' + props.userDetails._id, 'get')
+                .then((res) => {
+                    setAllProfiles(res.students)
+                })
+        }
+    }, [props]);
+
+    //Setting up display post API calls
+    useEffect(() => {
+        completeProfile();
+        constructDisplay();
+    }, [allProfiles]);
+
+    //Page change
+    useEffect(() => {
+        setPages(Math.ceil(filteredProfiles.length / pageSize));
+        constructDisplay()
+    }, [currPage, filteredProfiles]);
+
+    //Search change
+    useEffect(() => {
+        if (filter === 'all') {
+            setFilteredProfiles(allProfiles.filter((profile) => {
+                return profile.allText.includes(search.toLowerCase());
+            }));
+        } else if (filter === 'School:') {
+            setFilteredProfiles(allProfiles.filter((profile) => {
+                return (profile.allText.includes(search.toLowerCase()) 
+                    && profile.school.name.includes(secondaryFilter));
+            }));
+        } else if (filter === 'Grade:') {
+            setFilteredProfiles(allProfiles.filter((profile) => {
+                return (profile.allText.includes(search.toLowerCase()) 
+                    && profile.grade === secondaryFilter);
+            }));
+        } else if (filter === 'Grad Year:') {
+            setFilteredProfiles(allProfiles.filter((profile) => {
+                return (profile.allText.includes(search.toLowerCase()) 
+                    && profile.gradYear === secondaryFilter);
+            }));
+        }
+    }, [search, secondaryFilter])
+
+    //Filter category change
+    useEffect(() => {
+        if (filter === 'School:') {
+            setSecondaryFilterOptions(schoolOptions)
+        } else if (filter === 'Grad Year:') {
+            setSecondaryFilterOptions(gradYearOptions)
+        } else if (filter === 'Grade:') {
+            setSecondaryFilterOptions(gradeOptions)
+        }
+        setSecondaryFilter('')
+    }, [filter])
+
+    /* Helper functions */
+    const completeProfile = () => {
+        for (let profile of allProfiles) {
+            if (props.viewing === 'ALUMNI') {
+                profile.allText = (
+                    profile.city + ' ' + profile.country + ' '
+                    + profile.jobTitleName + ' ' + profile.companyName + ' '
+                    + profile.name + ' ' + profile.gradYear + ' '
+                    + profile.school.name
+                ).toLowerCase();
+            } else if (props.viewing === 'STUDENT') {
+                profile.allText = (
+                    profile.name + ' ' + profile.grade + ' '
+                    + profile.school.name
+                ).toLowerCase();
+            }
+        }
+        setFilteredProfiles(allProfiles)
+    }
+
+    const constructDisplay = () => {
+        if (filteredProfiles.length === 0 || filteredProfiles === null) return;
```

Comment:
This will throw an error is filteredProfiles is null so let's do `(!filteredProfiles || !filteredProfiles.length)`